File: context/example2.php
// example2.php
require_once "context.php";

function hello() {
    ctx_log_start("saying hello");
    printf("hello ");
    sleep(1);
    ctx_log_endok();
}

function greet($name) {
    ctx_log_start("greeting $name");
    printf("$name\n");
    sleep(2);
    ctx_log_endok();
}

ctx_set_log("output.php.ctxt");
ctx_log_start("running program");
hello();
greet("world");
ctx_log_endok();

File: context/example2.py
from time import sleep
import context.api as ctx

@ctx.log("Printing hello")
def function_a():
    print("hello")
    sleep(0.5)

@ctx.log("Printing world")
def function_b():
    print("world")
    sleep(0.1)

@ctx.log("Main loop", bookmark=True)
def main():
    function_a()
    function_b()

if __name__ == "__main__":
    ctx.set_log("file://example.ctxt")
    main()
